pages/setup/[token]/directory-sync/[directoryId]/edit.tsx

```diff
@@ -0,0 +1,161 @@
+import type { NextPage, GetServerSidePropsContext } from 'next';
+import React from 'react';
+import { useRouter } from 'next/router';
+import toast from 'react-hot-toast';
+import Link from 'next/link';
+import { ArrowLeftIcon } from '@heroicons/react/24/outline';
+
+import jackson from '@lib/jackson';
+import { inferSSRProps } from '@lib/inferSSRProps';
+import classNames from 'classnames';
+
+const Edit: NextPage<inferSSRProps<typeof getServerSideProps>> = ({
+  directory: { id, name, log_webhook_events, webhook },
+}) => {
+  const router = useRouter();
+  const { token } = router.query;
+  const [directory, setDirectory] = React.useState({
+    name,
+    log_webhook_events,
+    webhook_url: webhook.endpoint,
+    webhook_secret: webhook.secret,
+  });
+
+  const [loading, setLoading] = React.useState(false);
+
+  const onSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setLoading(true);
+
+    const rawResponse = await fetch(
+      token ? `/api/setup/${token}/directory-sync/${id}` : `/api/admin/directory-sync/${id}`,
```

Comment:
Should we need really this condition here? I assume `/setup/[token]` won't make a request to the `/api/admin` at any point. The page is inside the path `/setup/[token]`, so there will be a token always otherwise it becomes a 404 by default. If I'm right, request you to check similar cases in other places as well.